core/src/main/codegen/templates/Parser.jj

```diff
@@ -8300,6 +8332,11 @@ MORE :
     < IDENTIFIER: <LETTER> (<LETTER>|<DIGIT>)* >
 }
 
+<DEFAULT, DQID, BTID, BQID> TOKEN :
+{
+    < PARAM_IDENTIFIER: "@" (["\u0041"-"\u005a", "\u0061"-"\u007a", "\u005f"]) (["\u0041"-"\u005a", "\u0061"-"\u007a", "\u005f", "\u0030"-"\u0039"])* >
```

Comment:
Is there a reason you don't use LETTER or DIGIT?  It seems like you could do "@" + (<LETTER>|<DIGIT>)*

Reply:
Two reasons.

1. It wasn't working.
2. The letters/digits used by the Calcite parser include more than those that are allowed by a Python identifier. The regex (in Python) should be "@(_|[A-Z]|[a-z])(\w)*". The first character must be there and not a number. In addition, Letter/Digit seem to have some other unicode symbols defined that aren't in this set.

Reply:
kk